robot/impl/resource_manager.go

```diff
@@ -184,15 +184,53 @@ func (manager *resourceManager) updateRemoteResourceNames(
 ) bool {
 	manager.logger.CDebugw(ctx, "updating remote resource names", "remote", remoteName, "recreateAllClients", recreateAllClients)
 	activeResourceNames := map[resource.Name]bool{}
-	newResources := rr.ResourceNames()
+	ms, err := rr.MachineStatus(ctx)
+
+	var newResources []resource.Status
+	switch {
+	case err == nil:
+		newResources = ms.Resources
+
+	// The connection to the remote is broken. In this case, we mark each resource node
+	// on this remote as disconnected but do not report any other changes.
+	case errors.Is(err, client.ErrDisconnected):
+		remoteGraph, err := manager.resources.SubGraphFrom(remoteName)
+		if err != nil {
+			manager.logger.Error(
+				"unable to lookup remote resources internally",
+				"remote", remoteName,
+				"error", err,
+			)
+			return true
+		}
+		for _, name := range remoteGraph.Names() {
+			gNode, ok := manager.resources.Node(name)
+			if !ok {
+				continue
+			}
+			gNode.SetDisconnected()
+		}
+		return false
+
+	// Failed to fetch resources for some other reason. In this case,
+	// proceed with remote resource update as if we have no resources.
+	default:
+		manager.logger.Errorw(
+			"failed to fetch resource",
+			"remote", remoteName,
+			"error", err,
+		)
```

Comment:
similar to https://github.com/viamrobotics/rdk/pull/4273/files#r1773984731

this sort of mimics the current behavior (empty/nil resource names could be the result of an error), but really i think we should start bubbling up errors.